language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/CodeGenRunner.scala

```diff
@@ -91,13 +79,12 @@ private final class CodeGenRunner(
         Future.unit
     }
 
-  private def createTypeDefinitionClasses(
-      moduleWithContext: ModuleWithContext
-  ): Iterable[JavaFile] = {
-    MDC.put("packageId", moduleWithContext.packageId)
-    MDC.put("packageIdShort", moduleWithContext.packageId.take(7))
-    MDC.put("moduleName", moduleWithContext.name)
-    val typeSpecs = moduleWithContext.typesLineages.flatMap(ClassForType(_, packagePrefixes))
+  private def createTypeDefinitionClasses(module: ModuleWithContext): Iterable[JavaFile] = {
+    MDC.put("packageId", module.packageId)
+    MDC.put("packageIdShort", module.packageId.take(7))
+    MDC.put("moduleName", module.name)
+    val relevant = scope.serializableTypes.view.map(_._1).toSet
```

Comment:
This can probably be computed once and re-used.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/13571/commits/9fb5d981e35f32e6a284d2ce93c0d8f203273121